tools/c7n_gcp/c7n_gcp/resources/gke.py

```diff
@@ -11,19 +11,77 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import re
+
 from c7n_gcp.provider import resources
-from c7n_gcp.query import QueryResourceManager, TypeInfo
+from c7n_gcp.query import (QueryResourceManager, TypeInfo, ChildTypeInfo,
+                           ChildResourceManager)
+from c7n.utils import local_session
 
 
 @resources.register('gke-cluster')
 class KubernetesCluster(QueryResourceManager):
 
     class resource_type(TypeInfo):
         service = 'container'
-        version = 'v1beta1'
+        version = 'v1'
         component = 'projects.locations.clusters'
         enum_spec = ('list', 'clusters[]', None)
         scope = 'project'
         scope_key = 'parent'
         scope_template = "projects/{}/locations/-"
         id = "name"
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_query(
+                'get', verb_arguments={
+                    'name': 'projects/{}/locations/{}/clusters/{}'.format(
+                        resource_info['project_id'],
+                        resource_info['location'],
+                        resource_info['cluster_name'])})
+
+
+@resources.register('gke-cluster-nodepool')
+class KubernetesClusterNodePool(ChildResourceManager):
+
```

Comment:
please add class doc strings with the url to the gcp rest api docs for the resource

Reply:
Updated